internal/controller/isbservicerollout/isbservicerollout_controller.go

```diff
@@ -416,10 +416,10 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 			if err := riders.UpdateRidersInK8S(ctx, newISBServiceDef, riderAdditions, riderModifications, riderDeletions, r.client); err != nil {
 				return 0, err
 			}
-		}
 
-		// update the list of riders in the Status
-		r.SetCurrentRiderList(isbServiceRollout, currentRiderList)
+			// update the list of riders in the Status
+			r.SetCurrentRiderList(ctx, isbServiceRollout, currentRiderList)
```

Comment:
Originally, I was always calling this function no matter what. But I realized this should only be called after we've successfully called `UpdateRidersInK8S()`. 

This Status is supposed to represent what's truly on the cluster. If `UpdateRidersInK8S()` fails to make the update, and then we set this anyway to reflect what we intended to do, it causes issues on a future reconciliation when we are using this Status to determine what exists.